File: api/src/models/branch.ts
/**
 * @swagger
 * components:
 *   schemas:
 *     Branch:
 *       type: object
 *       required:
 *         - branchId
 *         - name
 *         - headquartersId
 *       properties:
 *         branchId:
 *           type: integer
 *           description: The unique identifier for the branch
 *         headquartersId:
 *           type: integer
 *           description: The ID of the headquarters this branch belongs to
 *         name:
 *           type: string
 *           description: The name of the branch
 *         address:
 *           type: string
 *           description: Physical address of the branch
 *         phone:
 *           type: string
 *           description: Contact phone number for the branch
 *         email:
 *           type: string
 *           format: email
 *           description: Contact email for the branch
 */
export interface Branch {
  branchId: number;
  headquartersId: number;
  name: string;
  description: string;
  address: string;
  contactPerson: string;
  email: string;
  phone: string;
}

File: api/src/main/java/com/github/av2/api/model/Branch.java
package com.github.av2.api.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Branch information")
public class Branch {
    @Schema(description = "The unique identifier for the branch", required = true)
    private Integer branchId;
    
    @Schema(description = "The ID of the headquarters this branch belongs to", required = true)
    private Integer headquartersId;
    
    @Schema(description = "The name of the branch", required = true)
    private String name;
    
    @Schema(description = "Description of the branch")
    private String description;
    
    @Schema(description = "Physical address of the branch")
    private String address;
    
    @Schema(description = "Contact person for the branch")
    private String contactPerson;
    
    @Schema(description = "Contact email for the branch")
    private String email;
    
    @Schema(description = "Contact phone number for the branch")
    private String phone;
}